woocommerce/woocommerce-ios
# Supported zero sale price in Edit Products

Fixes #2021 

## Description
While working on this I discovered that sending a `sale_price` equal to `0`, returns always an empty string. So there are two cases where `sale_price` contains an empty string:
1) The sale price is not configured
2) The sale price is `0`

So, the method to check correctly if there is a sale price is to check if `on_sale` is `true`.
To fix this issue that is not handled by the backend at the moment, I edited the `Product` model to decode correctly the sale price setting it to `0` when the product is on sale, and the `sale_price` is equal to empty string.

## Testing
1. Configure the sale price of a product from your website to zero
2. Launch the app and navigate to the product detail
3. Make sure the sale price showed is `0.00`
4. Try to edit the sale price, removing it.
5. Make sure that in product detail the sale price is no more showing.
6. Try to update the product. Make sure that in product detail the sale price is no more showing.
7. Try to repeat the procedure, with a sale price, like `10.00`.
8. Try to update the product, and make sure everything is ok.

Update release notes:
